Src/FluentAssertions/AssertionExtensions.cs

```diff
@@ -642,6 +642,16 @@ public static AsyncFunctionAssertions Should(this Func<Task> action)
             return new AsyncFunctionAssertions(action, extractor);
         }
 
+        /// <summary>
+        /// Returns a <see cref="AsyncFunctionAssertions"/> object that can be used to assert the
+        /// current <see><cref>System.Func{Task{T}}</cref></see>.
+        /// </summary>
+        [Pure]
+        public static AsyncFunctionAssertions Should<T>(this Func<Task<T>> action)
```

Comment:
So the overload that takes a `Func<T>` does not override this one?

Reply:
No, it does not.

For example all tests for `Should<T>(this Func<T> func)` and `Should(this Func<Task> action)` were not affected and still call respective overload. New overload is called only for two new tests I 've added.

Another important bit is `public class Task<TResult> : System.Threading.Tasks.Task`. Generic version inherits non-generic one, that allows overload resolution to choose `Task<T>` overload as one that is more specific.